qiskit_ionq/ionq_job.py

```diff
@@ -180,18 +176,22 @@ def __init__(
 
         if passed_args is not None:
             self.extra_query_params = passed_args.pop("extra_query_params", {})
-            self.extra_metadata = {
-                **passed_args.pop("extra_metadata", {}),
-                **(circuit.metadata or {}),
-            }
+            self.extra_metadata = passed_args.pop("extra_metadata", {})
             self._passed_args = passed_args
         else:
             self.extra_query_params = {}
             self.extra_metadata = {}
             self._passed_args = {"shots": 1024, "sampler_seed": None}
 
+        # Handle both single and list of circuits
         if circuit is not None:
-            self.circuit = circuit
+            if isinstance(circuit, (list, tuple)):
+                self.circuit = circuit
+                for circ in circuit:
+                    self.extra_metadata.update(circ.metadata or {})
+            else:
+                self.circuit = circuit
+                self.extra_metadata.update(circuit.metadata or {})
```

Comment:
This means that metadata from a circuit later in the list will override metadata earlier , yeah?

Do we want to warn/fail if that happens?

Reply:
Hope it's ok for me to comment. From my POV as a user, I rely on accurate preservation of the circuit metadata all the way through to the results, as a feature of `qiskit_ibm_runtime`. Here's my use case.

Since queue times can be hours or days, I run a Python process to launch a multicircuit job, then I terminate it so it doesn't block during queueing time. Periodically, I check in manually on the job completion status. Then when the job completes, I run a separate Python process to analyse the results.

Now, how to keep track of the functionality of the various circuits in the job? Locally storing metadata and correlating it by job ID and circuit index strikes me as a fragile solution. Instead, I rely on the circuit metadata being accurately passed through to the results. Then each circuit result has the original metadata already attached, and I just parse it as part of the result object.

Reply:
thanks

> Hope it's ok for me to comment. From my POV as a user, I rely on accurate preservation of the circuit metadata all the way through to the results, as a feature of `qiskit_ibm_runtime`. Here's my use case.
> 
> Since queue times can be hours or days, I run a Python process to launch a multicircuit job, then I terminate it so it doesn't block during queueing time. Periodically, I check in manually on the job completion status. Then when the job completes, I run a separate Python process to analyse the results.
> 
> Now, how to keep track of the functionality of the various circuits in the job? Locally storing metadata and correlating it by job ID and circuit index strikes me as a fragile solution. Instead, I rely on the circuit metadata being accurately passed through to the results. Then each circuit result has the original metadata already attached, and I just parse it as part of the result object.

thats great context for using metadata :) thank you! ill try to fix this in the pr, otherwise itll come in a patch release soon.

Reply:
Thanks for your kind response @splch. Can I ask about the timeline for this PR? Multicircuit job submission is currently a blocker for Q-CTRL's collaboration with John Gamble's team.